src/memory_tracker-inl.h

```diff
@@ -241,6 +241,36 @@ void MemoryTracker::Track(const MemoryRetainer* retainer,
   PopNode();
 }
 
+void MemoryTracker::TrackInlineField(const MemoryRetainer* container,
+                                     const MemoryRetainer* retainer,
+                                     const char* edge_name) {
+  v8::HandleScope handle_scope(isolate_);
+  auto it = seen_.find(retainer);
+  if (it != seen_.end()) {
+    if (CurrentNode() != nullptr) {
+      graph_->AddEdge(CurrentNode(), it->second, edge_name);
+    }
+    return;  // It has already been tracked, no need to call MemoryInfo again
+  }
+
+  // Get the node of the container this retainer is embedded in.
+  auto itc = seen_.find(container);
+
+  // It is an error not to have a container for inline fields.
+  CHECK(itc != seen_.end());
+  MemoryRetainerNode* n = PushNode(retainer, edge_name);
+  retainer->MemoryInfo(this);
+  CHECK_EQ(CurrentNode(), n);
+
+  // Adjust the size of the container as now we (the fields) are tracking
+  // ourselves.
+  MemoryRetainerNode* nc = itc->second;
+  nc->size_ -= n->size_;
+
+  CHECK_NE(n->size_, 0);
+  PopNode();
+}
```

Comment:
This seems to duplicate quite a bit from `Track()` – trying to compare the two, it looks like this is equivalent to this?

```c++
void MemoryTracker::TrackInlineField(const MemoryRetainer* retainer,
                                     const char* edge_name) {
  Track(retainer, edge_name);
  CurrentNode()->size_ -= retainer->SelfSize();
}
```

(`CurrentNode()` should be the same as `nc` in this code, if I am not mistaken?)

Reply:
@addaleax - you are right, that works perfectly to the purpose. I followed your code, with an additional `CHECK` on the current node.